openquake/commonlib/parallel.py

```diff
@@ -35,20 +35,34 @@
 from openquake.baselib.general import split_in_blocks, AccumDict, humansize
 from openquake.hazardlib.gsim.base import GroundShakingIntensityModel
 
-
 executor = ProcessPoolExecutor()
 # the num_tasks_hint is chosen to be 2 times bigger than the name of
 # cores; it is a heuristic number to get a good distribution;
 # it has no more significance than that
 executor.num_tasks_hint = executor._max_workers * 2
 
+OQ_DISTRIBUTE = os.environ.get('OQ_DISTRIBUTE', 'futures').lower()
+
+
+if OQ_DISTRIBUTE == 'celery':
```

Comment:
To be able to use `celery` the celeryconfig must be imported before any import from celery:

```python
try:
    import celeryconfig
except ImportError:
     sys.path.append('/usr/share/openquake/engine')
```